src/gravel/controllers/resources/inventory.py

```diff
@@ -37,10 +50,26 @@ async def probe(self) -> None:
         diff: int = int(time.monotonic()) - start
         logger.info(f"=> inventory probing took {diff} seconds")
         self._latest = nodeinfo
+        await self._publish()
 
-    async def latest(self) -> Optional[NodeInfoModel]:
+    @property
+    def latest(self) -> Optional[NodeInfoModel]:
         return self._latest
 
+    def subscribe(
+        self,
+        cb: Callable[[NodeInfoModel], Awaitable[None]],
+        once: bool
+    ) -> None:
+        self._subscribers.append(Subscriber(cb=cb, once=once))
+
+    async def _publish(self) -> None:
+        assert self._latest
```

Comment:
Probably should use self.latest?

Reply:
Personally I'm a fan of keeping assertions on internal state separate from properties. As I see it, properties are for the benefit of consumers; in this case we are returning `self._latest` on the property, as is, but we could, technically, be doing other foo that could, possibly, not be applicable to this assertion.

Reply:
Depends if we expect the method providing the property to change. However, this is an assert so that's reasonable to me.